copy as markdown, ask to chatgpt or claude

mael-bomane opened the pull request:
### Problem

https://x.com/jacobvcreech/status/1946231295698235570

### Summary of Changes

added `@radix-ui/react-dropdown-menu": "^2.1.15"` to `package.json`
added `@/app/components/ui/dropdown-menu.tsx`
added `@/app/components/copy-page.tsx`

integration of  `@/app/components/copy-page.tsx` in `@/app/components/docs-page.tsx`

<img width="313" height="289" alt="2025-07-18_21:54:01" src="https://github.com/user-attachments/assets/3781838d-cb95-400b-af57-496c16bac62c" />

Fixes #

GuiBibeau commented on src/app/components/copy-page.tsx:
```diff
@@ -0,0 +1,186 @@
+        <Button
+          variant="outline"
+          className="flex-1 items-center p-2 bg-popover rounded-r-none"
+          onClick={async () => {
```
please move the handler to the body of the component for easier future edits

mael-bomane replied:
indeed, it's repeated twice in the code, i'll do with `useCallback`

GuiBibeau commented on src/app/components/copy-page.tsx:
```diff
@@ -0,0 +1,186 @@
+  href,
+}: {
+  filePath: string;
+  href: string;
```
let's name this url instead of href.

GuiBibeau commented on src/app/components/copy-page.tsx:
```diff
@@ -0,0 +1,186 @@
+          className="flex-1 items-center p-2 bg-popover rounded-r-none"
+          onClick={async () => {
+            {
+              /* we fetch from client, to avoid server-side costs */
```
What does server side cost mean here? Are you reffering to a next.js server action?

mael-bomane replied:
I'm referring to pulling the markdown file content server-side from `@/content` at repo top-level, instead I'm fetching data client side (user cpu & network).

GuiBibeau replied:
Great! I appreciate it. I can see a lot of benefit in extracting this into a reusable function if you ever have time

mael-bomane replied:
Will do another PR for this, as this isn't directly related

GuiBibeau commented on src/app/components/copy-page.tsx:
```diff
@@ -0,0 +1,186 @@
+            );
+            if (response.ok) {
+              const text = await response.text();
+              console.log(text);
```
Let's remove the console.log

GuiBibeau commented on src/app/components/copy-page.tsx:
```diff
@@ -0,0 +1,186 @@
+              console.log(text);
+              navigator.clipboard.writeText(text);
+            } else {
+              console.log("error getting file");
```
should we handle that error with a toast or communicate the error message?

GuiBibeau commented on src/app/components/copy-page.tsx:
```diff
@@ -0,0 +1,186 @@
+          className="cursor-pointer focus:outline-none focus:ring-0"
+          onClick={async () => {
+            {
+              /* we fetch from client, to avoid server-side costs */
```
same question as above regarding what the comment means by server-side costs

GuiBibeau commented on src/app/components/copy-page.tsx:
```diff
@@ -0,0 +1,186 @@
+            <div className="flex items-center w-full">
+              <Button variant="ghost" size="icon" disabled>
+                <svg fill="currentColor" viewBox="0 0 24 24">
+                  <path d="M22.269 19.385H1.731a1.73 1.73 0 0 1-1.73-1.73V6.345a1.73 1.73 0 0 1 1.73-1.73h20.538a1.73 1.73 0 0 1 1.73 1.73v11.308a1.73 1.73 0 0 1-1.73 1.731zm-16.5-3.462v-4.5l2.308 2.885 2.307-2.885v4.5h2.308V8.078h-2.308l-2.307 2.885-2.308-2.885H3.461v7.847zM21.231 12h-2.308V8.077h-2.307V12h-2.308l3.461 4.039z" />
```
@h4rkl how do yo want to handle this svg for scalability? should this be moved to another folder of icons?

mael-bomane replied:
I did not wanted to add external npm package for icons (eg. `react-icons` or `font-awesome`) to keep it simple and making change to `package.json` as minimal as possible, with build size and attack surface in mind.

mael-bomane replied:
I created a new `@/app/components/icons/` folder, for custom jsx svg components (effectively extracting them from `@/app/components/copy-page.tsx` which wasn't clean.

To explain my decision : 

- `react-icons` is 86.2 MB
- `simple-icons` (smaller) is 15.7 MB
- importing any of these libraries would result in longer builds & loading times
- cannot style using svg `fillColor` property (light/dark theme) using `Image from 'next/image'`
- the 3 svg used in this case results in 3.58 KB

GuiBibeau commented on src/app/components/copy-page.tsx:
```diff
@@ -0,0 +1,186 @@
+                <span>
+                  View as MarkDown <ArrowUpRightIcon />
+                </span>
+                <span>View this page as plain text</span>
```
please move this to translation files to have it translated

GuiBibeau commented on src/app/components/copy-page.tsx:
```diff
@@ -0,0 +1,186 @@
+        {/* link with content encoded for ChatGPT */}
+        <DropdownMenuItem className="cursor-pointer focus:outline-none focus:ring-0">
+          <Link
+            href={`https://chat.openai.com/?hints=search&q=${encodeURIComponent(
+              `read from https://solana.com/docs/${filePath.replace("index.mdx", "").replace(".mdx", "")}\n so I can ask questions about it.`,
+            )}`}
```
let's move this as a variable in the body of the componet.

GuiBibeau commented on src/app/components/copy-page.tsx:
```diff
@@ -0,0 +1,186 @@
+                  xmlns="http://www.w3.org/2000/svg"
+                  className="w-4 h-4 shrink-0"
+                >
+                  <title>OpenAI</title>
```
let's move to translation files.

GuiBibeau replied:
Yes correct here 👍 resolving

GuiBibeau reviewed (changes requested):
thanks a lot for the contribution! The few changes i would do are summarized as:

- move text to translation files where we can for future translations
- move some logic a bit higher in the component body as a lot of it is straight in the rendering, making it hard to do future edits
- I'd love for @h4rkl to give see how to handle the svg as doing them inline makes discovery of future svg hard

tobySolutions commented:
s/o to @GuiBibeau. I think this an already good review.

mael-bomane commented:
# Summary of Changes

- handler top component using `useCallback`
- `href` renamed to `url`
- cleaner props typing
- removed comment on server-side fetching
- removed `console.log`
- added `sonner` toast to `layout.tsx`
- toasting error
- moved svg's to `@/app/components/icons`
- now with translations in `en/common.json`
- prompt moved to top component, also translated

h4rkl commented:
Cheers for the callout @GuiBibeau. If we can move those SVGs to `assets` then use `import <ICON> from "../../../assets/icons/<ICON>.svg";` that would be ideal. I realise that SVGs and PNGs are a bit of a mess in there but it's on the list to clean up and if we can keep things roughly in the same buckets it will make any refactor easier.
